pkg/cloudprovider/routecontroller/routecontroller.go

```diff
@@ -103,24 +99,18 @@ func (rc *RouteController) reconcile(nodes []api.Node, routes []*cloudprovider.R
 			// Check if this route applies to a node we know about & has correct CIDR.
 			if nodeCIDRs[route.TargetInstance] != route.DestinationCIDR {
 				// Delete the route.
-				go func(routeName string) {
-					if err := rc.routes.DeleteRoute(routeName); err != nil {
-						glog.Errorf("Could not delete route %s: %v", routeName, err)
+				routeCopy := *route
+				go func(route *cloudprovider.Route) {
+					if err := rc.routes.DeleteRoute(rc.clusterName, route); err != nil {
+						glog.Errorf("Could not delete route %s %s: %v", route.Name, route.DestinationCIDR, err)
 					}
-				}(route.Name)
+				}(&routeCopy)
```

Comment:
I don't think you need the routeCopy here for the go func. You could pass route directly to the function because the value of the pointer is captured right then (even though the pointer is reused).